src/pb_stub.cc

```diff
@@ -945,17 +945,19 @@ void
 Stub::Finalize()
 {
   finalizing_ = true;
-  // Stop async event loop if created.
-  if (!py::isinstance<py::none>(async_event_loop_)) {
-    async_event_loop_.attr("stop")();
-  }
-  // Call finalize if exists.
-  if (initialized_ && py::hasattr(model_instance_, "finalize")) {
-    try {
-      model_instance_.attr("finalize")();
+  if (initialized_) {
+    // Stop async event loop if created.
+    if (!py::isinstance<py::none>(async_event_loop_)) {
```

Comment:
Tangentially related question - is there a reasoning for not initializing all the `py::object` member variables to `py::none()` in the class declaration given that we know `py::object` [default construction](https://pybind11.readthedocs.io/en/stable/advanced/pycpp/object.html#gotchas) leads to possibly unexpected behavior as `nullptr`s?

```cpp
class Stub {
...
private:
...
  py::object model_instance_ = py::none();
  py::object deserialize_bytes_ = py::none();
  py::object serialize_bytes_ = py::none();
  py::object async_event_loop_ = py::none();
}
```

This would also avoid the unexpected entering of this condition.